bin/reth/src/node/mod.rs

```diff
@@ -213,9 +214,14 @@ fn network_config<DB: Database>(
     db: Arc<DB>,
     chain_id: u64,
     genesis_hash: H256,
+    config: &Config,
```

Comment:
Would like a network_config(db, genesis) method on Config which returns a Network_config if possible so we don't have many free functions like this.

Reply:
Would also like the NetworkConfig struct to have a spawn_network() method which does what start network does below instead of it being a free fn

Reply:
Added both